### DISCUSSION TITLE
General usage question

### DISCUSSION THREAD

**Discussion by trax72** [ManimCommunity/manim#2486] [Q&A] [answered]
Hi,

I need to animate some network graph algorithms with several hundreds of vertices and thousands of edges.

What I would like to do is draw the network graph and add/remove vertices/edges, change sizes, colors, labels and such during the runtime of the algorithm.

1. Any idea if this framework would be suitable for such number of objects (assuming each vertex and edge is an object)?

2. Is it possible to display the animation while the algorithm is running or is the only option to create an mpeg as output and display it later?

Thanks,
Tal.

**Comment by behackl** [comment, marked as answer]
To give you an idea of what you might have to expect: This video illustrating a depth-first search tree on a graph with 200 vertices and an edge density of 0.25 renders in medium quality in a bit less than 20sec.

https://user-images.githubusercontent.com/11851593/150773284-5c0e773f-a478-40f0-86ad-2703fedb7eec.mp4

Manim is not *great* for scenes like that, but it can work.

Regarding interactivity: In principle this is possible when using the (non-default) `opengl` renderer, but you might run into a bunch of bugs when doing so.
